cvat-ui/src/actions/models-actions.ts

```diff
@@ -64,34 +91,87 @@ export const modelsActions = {
             taskInstance,
         })
     ),
+    getModelProviders: () => createAction(ModelsActionTypes.GET_MODEL_PROVIDERS),
+    getModelProvidersSuccess: (providers: ModelProvider[]) => (
+        createAction(ModelsActionTypes.GET_MODEL_PROVIDERS_SUCCESS, {
+            providers,
+        })),
+    getModelProvidersFailed: (error: any) => createAction(ModelsActionTypes.GET_MODEL_PROVIDERS_FAILED, { error }),
+    getModelPreview: (modelID: string) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW, { modelID })
+    ),
+    getModelPreviewSuccess: (modelID: string, preview: string) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW_SUCCESS, { modelID, preview })
+    ),
+    getModelPreviewFailed: (modelID: string, error: any) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW_FAILED, { modelID, error })
+    ),
 };
 
 export type ModelsActions = ActionUnion<typeof modelsActions>;
 
 const core = getCore();
 
-export function getModelsAsync(): ThunkAction {
+export function getModelsAsync(query?: ModelsQuery): ThunkAction {
     return async (dispatch): Promise<void> => {
-        dispatch(modelsActions.getModels());
+        dispatch(modelsActions.getModels(query));
+
+        const filteredQuery = { ...query };
+        if (query) {
+            for (const key of Object.keys(query)) {
+                if ((filteredQuery as Indexable)[key] === null) {
+                    delete (filteredQuery as Indexable)[key];
+                }
+            }
+        }
```

Comment:
Can we write a function to remove null values from objects and use it for all the pages (projects, tasks, models, etc..)?